pkg/processor/timeout/timeout.go

```diff
@@ -134,19 +120,21 @@ func (w EventTimeoutWatcher) gracefulShutdown(ctx context.Context, timedoutWorke
 	w.shuttingDown = true
 
 	w.logger.WarnWithCtx(ctx, "Stopping triggers")
-	runningWorkers := w.stopTriggers(ctx, timedoutWorker)
+	busyTriggers := w.stopTriggers(ctx, timedoutWorker)
```

Comment:
I noticed the `gracefulShutdown` function is called in a goroutine when restarting a single worker.
What if this is called from multiple goroutines (for multiple workers) simultaneously?
Note that this flow is not only relevant for the Async flow, so we need to ensure no regression.

Reply:
@TomerShor 

> What if this is called from multiple goroutines (for multiple workers) simultaneously?

gracefulShutdown() can be called only from one place. Not sure how this pr can influence regression as I don't change the logic of calling gracefulShutdown..